lib/trace.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+var Stream = require('stream')
```

Comment:
I prefer using the [readable-stream](https://www.npmjs.com/package/readable-stream) module nowadays. It's a userland version of the stream module that ships with node and is maintained by of official Node.js Streams Working Group. It means they can fix bugs more easily and remove stuff only needed for backwards compatibility. Don't be fooled by the name of the module though - it's for all kind of streams, not only readable 😄